sdk/search/azure-search-documents/azure/search/documents/_internal/_search_index_document_batching_client.py

```diff
@@ -19,11 +19,37 @@
 from .._headers_mixin import HeadersMixin
 from .._version import SDK_MONIKER
 
+try:
+    ABC = abc.ABC
+except AttributeError:  # Python 2.7, abc exists, but not ABC
+    ABC = abc.ABCMeta("ABC", (object,), {"__slots__": ()})  # type: ignore
+
 if TYPE_CHECKING:
     # pylint:disable=unused-import,ungrouped-imports
     from typing import Any, Union
     from azure.core.credentials import AzureKeyCredential
 
+class PersistenceBase(ABC):
```

Comment:
Should we do a typing.Protocol instead?

Reply:
Seems like it is supported since Python 3.8

Reply:
https://github.com/Azure/azure-sdk-for-python/blob/5be5cf8d386eabbb55620f9fd71f706e4bbbb951/sdk/core/azure-core/azure/core/credentials.py#L12

should do the trick